### ISSUE TITLE
Cannot install package in a docker image

### ISSUE THREAD

**Issue by Winggi** [codalab/codalab-worksheets#2339]
After I installed or upgraded a package, I cannot use it. For example:

`cl run -n env --request-docker-image qipeng/hotpotqa-base:gpu --request-network "pip install transformers"`
And it succeeded. Then I run my model, it seems that the package haven't installed:
`cl run --request-docker-image qipeng/hotpotqa-base:gpu :env :src 'python main.py xxxxxxxx'`
`ModuleNotFoundError: No module named 'transformers'`

Is there something wrong? I don't know how to solve it...

**Comment by candicegjing** [comment]
@Winggi Could you provide the uuid of both bundles that you had problems with?

**Comment by candicegjing** [comment]
I ran your command on my side, but I saw a different error. I'll investigate once I have your bundle uuids and update you asap.

**Comment by Winggi** [comment]
@candicegjing 
I create a worksheet that simply shows my error. The uuid is `0x3aaff0e1766d4f31ace71c10f5c7c71c`. 
Thanks so much!

**Comment by candicegjing** [comment]
@Winggi based on your two steps:
1. `cl run -n env "pip install transformers" --request-docker-image qipeng/hotpotqa-base:gpu --request-network `
1. `cl run :env :src 'python main.py xxxxxxxx' --request-docker-image qipeng/hotpotqa-base:gpu `

Could you try to create a docker image, say `qipeng/hotpotqa-base:image_dependency` directly to DockerHub instead of creating a bundle?
Then when you run the second step, try the following instead:
`cl run :src 'python main.py xxxxxxxx' --request-docker-image qipeng/hotpotqa-base:image_dependency`
Basically, try to build a customized image that contains the dependency you need in one Docker image and then refer it as the docker image dependency in `request-docker-image`.

**Comment by nelson-liu** [comment]
@Winggi containers are re-created before each run, and destroyed for each run—it isn't like a normal shell, where you can run a sequence of command and expect the state of the system to change. So, your first `pip install transformers` works fine, but the changes aren't saved to the host because containers are single-use. You're trying to use `:env` as a dependency, but the package itself probably isn't one of the bundle outputs—it's installed to the system pip packages directory.

A workaround is to just combine it into one command, i.e.:

`cl run --request-docker-image qipeng/hotpotqa-base:gpu :src 'pip install transformers; python main.py xxxxxxxx'`

**Comment by Winggi** [comment]
Got it! I created my new docker image and it works now.
